src/ray/core_worker/task_event_buffer.cc

```diff
@@ -100,98 +102,85 @@ void TaskEventBufferImpl::AddTaskEvent(rpc::TaskEvents task_events) {
   absl::MutexLock lock(&mutex_);
 
   auto limit = RayConfig::instance().task_events_max_num_task_events_in_buffer();
-  if (limit > 0 && buffer_.size() >= static_cast<size_t>(limit)) {
+  if (limit > 0 && data_->events_by_task_size() >= limit) {
     // Too many task events, start overriding older ones.
-    if (buffer_[next_idx_to_overwrite_].has_profile_events()) {
+    if (data_->events_by_task().at(next_idx_to_overwrite_).has_profile_events()) {
       num_profile_task_events_dropped_++;
     } else {
       num_status_task_events_dropped_++;
     }
 
-    buffer_[next_idx_to_overwrite_] = std::move(task_events);
+    // The two should be from different arena : so calling Swap will not improve
```

Comment:
what do you mean by "the two"?

Reply:
The new task_events and the one to be overwritten.